service/controller/resource/nodepool/template/main.json

```diff
@@ -284,6 +299,11 @@
         },
         "singlePlacementGroup": "[parameters('vmssSinglePlacementGroup')]",
         "virtualMachineProfile": {
+          "evictionPolicy": "Delete",
+          "priority": "[if(parameters('spotInstancesEnabled'), 'Spot', 'Regular')]",
+          "billingProfile": {
```

Comment:
will this be ignored if `priority` is `regular`?

Reply:
Documentation says:

> Specifies the eviction policy for the Azure Spot virtual machine and Azure Spot scale set.

So I assume it is ignored if you don't have spot instances.
Plus in my tests I didn't notice issues.